django/db/models/fields/__init__.py

```diff
@@ -1500,6 +1500,8 @@ def __init__(self, verbose_name=None, name=None, max_digits=None,
                  decimal_places=None, **kwargs):
         self.max_digits, self.decimal_places = max_digits, decimal_places
         super(DecimalField, self).__init__(verbose_name, name, **kwargs)
+        self.validators.append(validators.DecimalValidator(self.max_digits,
+                                                           self.decimal_places))
```

Comment:
I would only add this validator `if self.max_digits is not None and self.decimal_places is not None` instead of checking it in `DecimalValidator.__call__`.